smartcontractkit/external-adapters-js
# Catch unhandled promise rejections

## Description

When upgrading adapters to from Node version 14 to 17, behavior that would previously fail now additionally causes the process to exit. This PR adds a small global handler set up in the expose function of the core framework, to effectively simulate the old behavior without making changes to infrastructure.

## Changes

- Add global handler for unhandled promise rejections

## Steps to Test

1. Without these changes, add background process to any adapter, with a failing unhandled promise 
2. The rejection should cause the process to exit
3. Pull this PR, and apply the same changes as before
4. The error should still occur, but the process should stay alive

## Quality Assurance
